extensions/smallrye-fault-tolerance/runtime/src/main/java/io/quarkus/smallrye/faulttolerance/runtime/QuarkusFaultToleranceOperationProvider.java

```diff
@@ -1,22 +1,39 @@
 package io.quarkus.smallrye.faulttolerance.runtime;
 
 import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 import javax.annotation.Priority;
 import javax.enterprise.context.Dependent;
 import javax.enterprise.inject.Alternative;
 
+import org.jboss.logging.Logger;
+
 import io.smallrye.faulttolerance.FaultToleranceOperationProvider;
 import io.smallrye.faulttolerance.config.FaultToleranceOperation;
 
 @Dependent
 @Alternative
 @Priority(1)
 public class QuarkusFaultToleranceOperationProvider implements FaultToleranceOperationProvider {
+    private static final Logger LOG = Logger.getLogger(QuarkusFaultToleranceOperationProvider.class);
+    static final Map<String, FaultToleranceOperation> OPERATIONS = new ConcurrentHashMap<>();
 
     @Override
     public FaultToleranceOperation get(Class<?> beanClass, Method method) {
+        String key = getCacheKey(beanClass, method);
+        FaultToleranceOperation operation = OPERATIONS.get(key);
+        return operation != null ? operation : createAtRuntime(beanClass, method, key);
```

Comment:
We could use `Map.computeIfAbsent()` if a `FaultToleranceOperation` is not found in the cache...

Reply:
This was on my first implémentation, but as with the current one the operation should always be in the cache I don't know if it worth it.
It is here to catch potential bugs, and as it will be inside the hot code path I prefere to use `get` instead of `computeIfAbsent`. We could revisit it later if needed.

Reply:
My HV benchmarks taught me the hard way that using `get()` is faster if having the operation in the cache is the most common use case.

Reply:
+1

Reply:
I remember having similar discussions on the PRs :). I say no `computeIfAbsent` anywhere on the hot path :)

Reply:
Well, it depends on how many times it's invoked ;-). From my experience microbenchmarks are often tricky and the bottlenecks identified in the infra code might not be bottlenecks at all in a real app and vice versa.

Reply:
Well, with native image, stuff are a little different as there is no optimisation based on branch prediction (unless for paid customers using the new PGO mechanism of GraalVM). So computeIfAbsent could be more slow on native image than on JVM mode ...
Having said that, if I didn't saw any benchmark written with JMH that prove this I will deny it ;)

Anyway, I switch to regular HashMap and uses `get` as the operation should always be present (in fact I can move to a runtime exception if not in the cache if we prefere ... like a NPE :) )